llvm/lib/Target/AVR/AVRISelLowering.cpp

```diff
@@ -2521,13 +2521,13 @@ AVRTargetLowering::getConstraintType(StringRef Constraint) const {
   return TargetLowering::getConstraintType(Constraint);
 }
 
-unsigned
+InlineAsm::ConstraintCode
 AVRTargetLowering::getInlineAsmMemConstraint(StringRef ConstraintCode) const {
   // Not sure if this is actually the right thing to do, but we got to do
   // *something* [agnat]
   switch (ConstraintCode[0]) {
```

Comment:
I know this is just a simple refactoring in this case, but could you turn this into an if-then statement instead? The switch seems unnecessary.

Reply:
Disagree. This causes much less churn if and when new cases need to be added.

Reply:
Sorry @bwendling , I agree with @jrtc27 .  These kinds of memory-only constraints don't get added often, but I also doubt that LLVM supports all of https://gcc.gnu.org/onlinedocs/gcc/Machine-Constraints.html for all backend LLVM has.  I forsee these changing, at least for the more modern and well supported architectures.